src/OmniSharp.Roslyn/MetadataHelper.cs

```diff
@@ -58,8 +58,14 @@ public Task<Document> GetDocumentFromMetadata(Project project, ISymbol symbol, C
         {
             var filePath = GetFilePathForSymbol(project, symbol);
             var topLevelSymbol = GetTopLevelContainingNamedType(symbol);
-            var temporaryDocument = project.AddDocument(filePath, string.Empty);
 
+            // since submission projects cannot have new documents added to it
+            // we will use temporary project to hold metadata documents
+            var metadataProject = project.IsSubmission ? project.Solution.AddProject("metadataTemp", "metadataTemp.dll", LanguageNames.CSharp).
+                WithCompilationOptions(project.CompilationOptions).
+                WithMetadataReferences(project.MetadataReferences) : project;
```

Comment:
Tiny, tiny, tiny nit: This might be more readable if formatted like so:

C#
```
var metadataProject = project.IsSubmission
    ? project.Solution.AddProject("metadataTemp", "metadataTemp.dll", LanguageNames.CSharp)
        .WithCompilationOptions(project.CompilationOptions)
        .WithMetadataReferences(project.MetadataReferences)
    : project;
```